File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.js
/**
 * @param {string} instructions
 * @return {boolean}
 */
var isRobotBounded = function(instructions) {
  const dirs = [[0, 1], [1, 0], [0, -1], [-1, 0]];
  let [x, y, d] = [0, 0, 0];
  
  for (let ch of instructions) {
    if (ch === 'R') d = (d + 1) % 4;
    else if (ch === 'L') d = (d + 3) % 4;
    else {
      x += dirs[d][0];
      y += dirs[d][1];
    }
  }
  
  return !x && !y || d > 0;
};

File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        dirs = ((0, 1), (1, 0), (0, -1), (-1, 0))
        x = y = d = 0
        for ch in instructions:
            if ch == 'R':
                d = (d + 1) % 4
            elif ch == 'L':
                d = (d + 3) % 4
            else:
                x += dirs[d][0]
                y += dirs[d][1]
        return (x, y) == (0, 0) or d > 0